### PR TITLE
feat: adapt Kafka bindings to v3 - take 2

### PR DESCRIPTION
**Pull request by lbroudoux**
**Description**

- This is a replacement to #221 to better align on other bindings that move to v3
- See the discussion and decision summarized [here](https://github.com/asyncapi/bindings/issues/182#issuecomment-1825860815)

**Related issue(s)**

Relates to https://github.com/asyncapi/bindings/issues/182

### CODE REVIEW

File: kafka/README.md
Comment (dalelane) [inline kafka/README.md:1]: Changes look good. I think we also need to remove the comment on line 60 (as it's a copy/paste error from elsewhere on the page)

> This example is valid for any Confluent compatible schema registry. Here we describe the implementation using the first 4 bytes in payload to store schema identifier.

**Review by dalelane — APPROVED** [review]

**Comment by jonaslagoni** [comment]
As we are preparing to merge https://github.com/asyncapi/bindings/pull/224 ASAP, do you folks want to have this included or want to re-target the PR after merge? 😄

**Comment by dalelane** [comment]
sorry - I'd missed that the merge had dismissed my review

**Review by dalelane — DISMISSED** [review]

File: kafka/README.md
Comment (dalelane) [inline kafka/README.md:129]: This indenting doesn't look right. Is it supposed to be like this?

```suggestion
          type: string
```

File: kafka/README.md
Comment (dalelane) [inline kafka/README.md:133]:
```suggestion
          type: string
          enum: ['myClientId']
```

**Comment by dalelane** [comment]
@lbroudoux I think some indenting needs an update - I've put suggestions in. Do they look okay?

File: kafka/README.md
Comment (lbroudoux) [reply kafka/README.md:130]: Yes, you're right. I don't know what happened. Thank you for your vigilance.

**Comment by lbroudoux** [comment]
@dalelane Ok for the 2 suggested changes and I found some others that were also bad. Sorry for the mistakes and thanks for your vigilance. I think we're good to go now.

**Review by dalelane — APPROVED** [review]

**Pull request merged**
